src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -23,9 +26,10 @@ public class SystemLogger : ILogger
         private readonly IDebugStateProvider _debugStateProvider;
         private readonly IScriptEventManager _eventManager;
         private readonly IExternalScopeProvider _scopeProvider;
+        private readonly IOptionsMonitor<AppServiceOptions> _appServiceOptions;
 
         public SystemLogger(string hostInstanceId, string categoryName, IEventGenerator eventGenerator, IEnvironment environment,
-            IDebugStateProvider debugStateProvider, IScriptEventManager eventManager, IExternalScopeProvider scopeProvider)
+            IDebugStateProvider debugStateProvider, IScriptEventManager eventManager, IExternalScopeProvider scopeProvider, IOptionsMonitor<AppServiceOptions> appServiceOptions)
```

Comment:
Using these options, we can avoid recomputing/formatting the 4 app level aspects on each log statement, which involved a lot of string manipulations.